composition-js/src/merging/merge.ts

```diff
@@ -2045,10 +2055,14 @@ class Merger {
       const external = this.isExternal(idx, source);
       const sourceMeta = this.subgraphs.values()[idx].metadata();
       const name = this.joinSpecName(idx);
+      
+      // fields in this subgraph that are no longer viable because they've been overridden 
+      const overriddenFields = this.completelyOverriddenFieldMap.get(idx);
```

Comment:
Note that:
1. Since the result of overridden field removal may be to remove the `@provides` entirely, this could result in the `@join__field` becoming unnecessary. This means the computation of the "updated" `@provides` needs to be done before that, so that `needsJoinField()` can be aware of it. (Might want to put it in `FieldMergeContextProperties`.)
2. In `@join__field`, the `usedOverridden` argument is set based on whether the overridden field is "used" (i.e. is in `@key`/`@requires`/`@provides`/`@fromContext` or is an object field that implements some interface field in the subgraph). If it's used, then the field still needs to be in the overridden subgraph, but otherwise (for non-progressive overrides) it is effectively dropped from supergraph schema metadata.
    - This definition of "used" should account for the `@provides` effectively omitting (non-progressive) overridden fields. My suggestion here is that the `collectUsedFields()` function should collect two separate sets of `FieldDefinition`s: one for `@provides`, and one for the rest. Both of these should be stored in `FederationMetadata`.
    - The `FederationMetadata.isFieldUsed()` method should check both sets. You should add a new method to `FederationMetadata` called e.g. `isFieldUsedIgnoringProvides()` that only checks the non-`@provides` set.
    - In `validateOverride()`, when computing `overriddenFieldIsReferenced`, use `isFieldUsedIgnoringProvides()` instead of `isFieldUsed()` if `overrideLabel` is unset.